cloudinit/config/cc_growpart.py

```diff
@@ -210,11 +210,35 @@ def get_size(filename):
 
 def device_part_info(devpath):
     # convert an entry in /dev/ to parent disk and partition number
+    myenv = os.environ.copy()
+    myenv['LANG'] = 'C'
 
     # input of /dev/vdb or /dev/disk/by-label/foo
     # rpath is hopefully a real-ish path in /dev (vda, sdb..)
     rpath = os.path.realpath(devpath)
 
+    is_lvm = True if re.search('.*dm.*', rpath) else False
```

Comment:
I think use of device mapper doesn't necessarily mean lvm.
and fwiw, the above is the same as:

```python
is_lvm = 'dm' in rpath
```

Perhaps a better func would be:

```python
def is_lvm_lv(devpath):
    # all lvm lvs will have a realpath as a 'dm-*' name.
    rpath = os.path.realpath(devpath)
    if not os.path.basename(rpath).startswith("dm-"):
        return False
    out, _ = subp.subp("udevadm", "info", devpath)
    # lvs should have DM_LV_NAME=<lvmuuid> and also DM_VG_NAME
    return 'DM_LV_NAME=' in out
```

Reply:
this whole function should check for Linux first, and even then, you're not guaranteed to have udevadm, like on Alpine

Reply:
@igalic Hi. Alpine cloud-init package maintainer here - while Alpine normally uses mdev instead of udev, in the case of the cloud-init package I explicitly added a dependancy on eudev (Alpine's packaged version of udev) as a couple of the cloud-init modules (persistent net device naming & disk_setup) expect udev to be in use.

Packaged cloud-init dependancies visible here: https://pkgs.alpinelinux.org/package/edge/community/x86_64/cloud-init